collect_app/src/main/java/org/odk/collect/android/activities/FormEntryActivity.java

```diff
@@ -1255,7 +1055,234 @@ public void onClick(View v) {
+            sa.setVisibility(View.VISIBLE);
+            saveAs.setText(saveName);
+            saveAs.setEnabled(true);
+            saveAs.setVisibility(View.VISIBLE);
+        } else {
+            // if instanceName is defined in form, this is the name -- no
+            // revisions
+            // display only the name, not the prompt, and disable edits
+            TextView sa = (TextView) endView
+                    .findViewById(R.id.save_form_as);
+            sa.setVisibility(View.GONE);
+            saveAs.setText(saveName);
+            saveAs.setEnabled(false);
+            saveAs.setBackgroundColor(Color.WHITE);
+            saveAs.setVisibility(View.VISIBLE);
+        }
+
+        // override the visibility settings based upon admin preferences
+        if (!mAdminPreferences.getBoolean(
+                AdminPreferencesActivity.KEY_SAVE_AS, true)) {
+            saveAs.setVisibility(View.GONE);
+            TextView sa = (TextView) endView
+                    .findViewById(R.id.save_form_as);
+            sa.setVisibility(View.GONE);
+        }
+
+        // Create 'save' button
+        ((Button) endView.findViewById(R.id.save_exit_button))
+                .setOnClickListener(new OnClickListener() {
+                    @Override
+                    public void onClick(View v) {
+                        Collect.getInstance()
+                                .getActivityLogger()
+                                .logInstanceAction(
+                                        this,
+                                        "createView.saveAndExit",
+                                        instanceComplete.isChecked() ? "saveAsComplete"
+                                                : "saveIncomplete");
+                        // Form is marked as 'saved' here.
+                        if (saveAs.getText().length() < 1) {
+                            Toast.makeText(FormEntryActivity.this,
+                                    R.string.save_as_error,
+                                    Toast.LENGTH_SHORT).show();
+                        } else {
+                            saveDataToDisk(EXIT, instanceComplete
+                                    .isChecked(), saveAs.getText()
+                                    .toString());
+                        }
+                    }
+                });
+
+        if (mBackButton.isShown()) {
+            mBackButton.setEnabled(true);
+        }
+        if (mNextButton.isShown()) {
+            mNextButton.setEnabled(false);
+        }
+
+        return endView;
+    }
+
+    private View createFormBeginningView(FormController formController, boolean advancingPage) {
+        View startView = View.inflate(this, R.layout.form_entry_start, null);
+        setTitle(getString(R.string.app_name) + " > " + formController.getFormTitle());
+
+		if (shouldShowIntroScreenBecauseOfLogo(formController)) { // If image is specified, we will always show intro screen
```

Comment:
Let's follow the Android project style guide for now and [keep lines short](http://source.android.com/source/code-style.html#limit-line-length). For conditionals, that means commenting on the next line unless you just have something very short to say:
```
if () {
    // long comment here
    ...
} else {
    // long comment here
    ...
}
```